lib/ts/recipe/thirdpartyemailpassword/recipe.ts

```diff
@@ -0,0 +1,441 @@
+        return this.emailPasswordRecipe.signIn(email, password);
+    };
+
+    signInUp = async (
+        thirdPartyId: string,
+        thirdPartyUserId: string,
+        email: {
+            id: string;
+            isVerified: boolean;
+        }
+    ): Promise<{ createdNewUser: boolean; user: User }> => {
+        if (this.thirdPartyRecipe === undefined) {
+            throw new STError(
+                {
+                    type: STError.BAD_INPUT_ERROR,
+                    message: "No thirdparty provider configured",
+                },
+                this.getRecipeId()
+            );
+        }
+        return this.thirdPartyRecipe.signInUp(thirdPartyId, thirdPartyUserId, email);
+    };
+
+    getUserById = async (userId: string): Promise<User | undefined> => {
+        let user: User | undefined = await this.emailPasswordRecipe.getUserById(userId);
+        if (user !== undefined) {
+            return user;
+        }
+        if (this.thirdPartyRecipe === undefined) {
+            return undefined;
+        }
+        return await this.thirdPartyRecipe.getUserById(userId);
+    };
+
+    getUserByThirdPartyInfo = async (thirdPartyId: string, thirdPartyUserId: string): Promise<User | undefined> => {
+        if (this.thirdPartyRecipe === undefined) {
+            return undefined;
+        }
+        return this.thirdPartyRecipe.getUserByThirdPartyInfo(thirdPartyId, thirdPartyUserId);
+    };
+
+    getEmailForUserId = async (userId: string) => {
+        let userInfo = await this.getUserById(userId);
+        if (userInfo === undefined) {
+            throw new STError(
+                {
+                    type: STError.UNKNOWN_USER_ID_ERROR,
+                    message: "Unknown User ID provided",
+                },
+                this.getRecipeId()
+            );
+        }
+        return userInfo.email;
+    };
+
+    getUserByEmail = async (email: string): Promise<User | undefined> => {
+        return this.emailPasswordRecipe.getUserByEmail(email);
+    };
+
+    createResetPasswordToken = async (userId: string): Promise<string> => {
+        return this.emailPasswordRecipe.createResetPasswordToken(userId);
+    };
+
+    resetPasswordUsingToken = async (token: string, newPassword: string) => {
+        return this.emailPasswordRecipe.resetPasswordUsingToken(token, newPassword);
+    };
+
+    createEmailVerificationToken = async (userId: string): Promise<string> => {
+        return this.emailVerificationRecipe.createEmailVerificationToken(userId, await this.getEmailForUserId(userId));
+    };
+
+    verifyEmailUsingToken = async (token: string) => {
+        return this.emailVerificationRecipe.verifyEmailUsingToken(token);
+    };
+
+    isEmailVerified = async (userId: string) => {
+        return this.emailVerificationRecipe.isEmailVerified(userId, await this.getEmailForUserId(userId));
+    };
+
+    getUsersOldestFirst = async (limit?: number, nextPaginationToken?: string) => {
```

Comment:
CHange name to `nextPaginationTokenString`

Reply:
done